### PR TITLE
Added LoginController with sigin endpoint, moved AuthenticatioExcepti…

### PR DESCRIPTION
**Pull request by natam**
Added LoginController with sigin endpoint, moved AuthenticatioException handling to ControllerExceptionHandler class

### CODE REVIEW

**Review by faruka-startsteps — CHANGES_REQUESTED** [review]

File: src/main/java/com/nkh/ECommerceShop/security/jwt/AuthEntryPointJwt.java [hunk 1]
```diff
@@ -7,34 +7,30 @@
...
 import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerExceptionResolver;
 
 import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
 
 @Component
 public class AuthEntryPointJwt implements AuthenticationEntryPoint {
+  @Autowired
```
Comment (faruka-startsteps) [inline src/main/java/com/nkh/ECommerceShop/security/jwt/AuthEntryPointJwt.java:24]: constructor autowiring

File: src/main/java/com/nkh/ECommerceShop/security/jwt/AuthEntryPointJwt.java [hunk 2]
```diff
@@ -7,34 +7,30 @@
...
+  @Qualifier("handlerExceptionResolver")
```
Comment (faruka-startsteps) [inline src/main/java/com/nkh/ECommerceShop/security/jwt/AuthEntryPointJwt.java:25]: do you need the @Qualifier there?

**Review by faruka-startsteps — APPROVED** [review]
Feel free to merge after addressing the comments

File: src/main/java/com/nkh/ECommerceShop/security/jwt/AuthEntryPointJwt.java (hunk 2 above)
Comment (natam) [reply src/main/java/com/nkh/ECommerceShop/security/jwt/AuthEntryPointJwt.java:25]: yes. it was in 2 different tutorials and I get error on removing it.

File: src/main/java/com/nkh/ECommerceShop/security/jwt/AuthEntryPointJwt.java (hunk 1 above)
Comment (natam) [reply src/main/java/com/nkh/ECommerceShop/security/jwt/AuthEntryPointJwt.java:24]: fixed

**Pull request merged**